api/src/main/java/org/pentaho/platform/api/genericfile/IGenericFileService.java

```diff
@@ -291,11 +292,59 @@ default IGenericFile getFile( @NonNull String path ) throws OperationFailedExcep
   List<IGenericFile> getDeletedFiles() throws OperationFailedException;
 
   /**
-   * Permanently deletes files given their paths.
+   * Permanently deletes files, given their paths.
    *
    * @param paths The list of file paths to be permanently deleted.
+   * @throws AccessControlException        If the current user cannot perform this operation.
+   * @throws BatchOperationFailedException If the batch operation fails for some reason.
+   * @throws OperationFailedException      If the operation fails for some other (checked) reason.
+   */
+  void deleteFilesPermanently( @NonNull List<GenericFilePath> paths ) throws OperationFailedException;
+
+  /**
+   * Permanently deletes a file, given its path.
+   *
+   * @param path The file path to be permanently deleted.
    * @throws AccessControlException   If the current user cannot perform this operation.
    * @throws OperationFailedException If the operation fails for some other (checked) reason.
    */
-  void deleteFilesPermanently( @NonNull List<GenericFilePath> paths ) throws OperationFailedException;
+  void deleteFilePermanently( @NonNull GenericFilePath path ) throws OperationFailedException;
+
+  /**
+   * Deletes files by sending them to the trash, given their paths.
+   *
+   * @param paths The list of file paths to be deleted.
+   * @throws AccessControlException        If the current user cannot perform this operation.
+   * @throws BatchOperationFailedException If the batch operation fails for some reason.
+   * @throws OperationFailedException      If the operation fails for some other (checked) reason.
+   */
+  void deleteFiles( @NonNull List<GenericFilePath> paths ) throws OperationFailedException;
+
+  /**
+   * Deletes a file by sending it to the trash, given its path.
+   *
+   * @param path The file path to be deleted.
+   * @throws AccessControlException   If the current user cannot perform this operation.
+   * @throws OperationFailedException If the operation fails for some other (checked) reason.
+   */
+  void deleteFile( @NonNull GenericFilePath path ) throws OperationFailedException;
```

Comment:
I think we're missing documenting, at least, the `NotFoundException`. 

One obvious case this happens is if the operation is called with the path for a file which is already in the trash!